### PR TITLE
Signed distance field in the software renderer

### PR DESCRIPTION
**Pull request by ogoffart**
Note that this is still disabled in the compiler with no way to enable
it with public API

### CODE REVIEW

**Comment by ogoffart** [comment]
The screenshot test is failling because the CI passes `--all-features` which enabled the sdf rendering

**Comment by ogoffart** [comment]
Now I don't understand why the CI passed.

File: internal/core/graphics/image.rs [hunk 1]
```diff
@@ -234,6 +234,11 @@ pub enum PixelFormat {
     RgbaPremultiplied,
     /// Alpha map. 8bits. Each pixel is an alpha value. The color is specified separately.
     AlphaMap,
+    /// Distance field. 8bit interpreted as i8.
+    /// The range is such that i8::MIN corresponds to 3 pixels outside of the shape,
+    /// and i8::MAX corresponds to 3 pixels inside the shape.
+    /// The array must be width * height +1 bytes long. (the extra bit is read but never used)
+    SignedDistanceField,
```
Comment (tronical) [inline internal/core/graphics/image.rs:241]: In light of future multi-channel support, perhaps for color emojis, I wonder if we should call this `SingleChannelSignedDistanceField`? I guess we can rename it later as this is private API.

File: internal/core/graphics/image.rs [hunk 2]
```diff
@@ -473,6 +479,9 @@ impl ImageInner {
                                 });
                                 slice.fill_with(|| iter.next().unwrap());
                             }
+                            PixelFormat::SignedDistanceField => {
```
Comment (tronical) [inline internal/core/graphics/image.rs:482]: Do you know where this might be called from?

Comment (ogoffart) [reply internal/core/graphics/image.rs:482]: Currently this is not called, but could be called if we were to generate distance field for @image-url

**Pull request merged**